### PR TITLE
ENH: The unit about building and adding a new EMC method

### PR DESCRIPTION
**Pull request by oesteban**
Instead of dumping the documentation scattered around projects, let's have the conversation about EMC here.

@arokem and @dPys, can you please have a look and suggest changes/point out things that seem wrong or even blatantly wrong?

### CODE REVIEW

File: docs/dmriprep/hmc.md [hunk 1]
```diff
@@ -1 +1,132 @@
-# Head Motion Correction
+# Contributing to *dMRIPrep* - the case of head-motion and eddy-currents estimation
+
+All *NiPreps* are open to the community feedback and contribution.
+Collaborating to seemingly big projects (*dMRIPrep* is, in reality, pretty small) can feel scary at first.
```
Comment (arokem) [inline docs/dmriprep/hmc.md:4]:
```suggestion
Contributing to seemingly big projects (*dMRIPrep* is, in reality, pretty small) can feel scary at first.
```
Or (less preferred) "collaborating within" ?

File: docs/dmriprep/hmc.md [hunk 2]
```diff
@@ -1 +1,132 @@
...
+
+## Step 0: check the contributing guidelines
+Before contributing to an open-source initiative, your first move *should* be checking how and whether the project is seeking feedback from the outside.
+Typically, that is stated in the `CONTRIBUTING.md` file of the project or similar solution (e.g., all [*NiPreps* share contributing guidelines](https://nipreps.org/community/)).
+
+## Step 1: preparing a proposal for a new feature, some documentation, or a bugfix
+Impromptu contributions are typically difficult to adsorb by projects involving several developers and/or with many users.
```
Comment (arokem) [inline docs/dmriprep/hmc.md:11]:
```suggestion
Impromptu contributions are typically difficult to absorb by projects involving several developers and/or with many users.
```

File: docs/dmriprep/hmc.md [hunk 3]
```diff
@@ -1 +1,132 @@
...
+For this reason, please first navigate the project's home (https://github.com/nipreps/dmriprep).
```
Comment (arokem) [inline docs/dmriprep/hmc.md:12]:
```suggestion
For this reason, please first browse through the project's repository home page (https://github.com/nipreps/dmriprep).
```

File: docs/dmriprep/hmc.md [hunk 4]
```diff
@@ -1 +1,132 @@
...
+
+## Step 2: plan ahead
+Once you get the thumbs-up from the maintainers, it is time to sort out your contribution.
+
+In this book we will show how to integrate a new algorithm into *dMRIPrep*.
+First you should accurately define the problem:
+
+**Problem**: although most dMRI practitioners employ FSL for the estimation and correction of head-motion and eddy-current distortions, the FSL toolbox has restrictions of commercial use. By using FSL's eddy implementation, *dMRIPrep* inherits those restrictions.
+Therefore, a fully-open implementation of *dMRIPrep* would require eliminating FSL as a dependency.
+In this case, we will provide our own implementation of an algorithm for head-motion and eddy-current-derived distortions.
```
Comment (arokem) [inline docs/dmriprep/hmc.md:29]: I don't know if I like this framing. Arguably, we want to provide a general, open-source solution that is also best in class. The license issues, though real, seem a bit obscure.

File: docs/dmriprep/hmc.md [hunk 5]
```diff
@@ -1 +1,132 @@
...
+
+**The proposed solution**: we will design a SHORELine-like algorithm for the realignment of dMRI data.
```
Comment (arokem) [inline docs/dmriprep/hmc.md:31]:
```suggestion
**The proposed solution**: we will design a model-based algorithm for the realignment of dMRI data.
This is based on an idea first proposed by [Ben-Amitay et al.](https://pubmed.ncbi.nlm.nih.gov/22183784/) and implemented as the SHORELine algorithm in [qsiprep](https://qsiprep.readthedocs.io/en/latest/). 
```

File: docs/dmriprep/hmc.md [hunk 6]
```diff
@@ -1 +1,132 @@
...
+The idea works as follows:
+
+  1. Leave one DWI (diffusion weighted image) orientation out
```
Comment (arokem) [inline docs/dmriprep/hmc.md:34]:
```suggestion
  1. Leave one DWI (diffusion weighted image) orientation out.
```

File: docs/dmriprep/hmc.md [hunk 7]
```diff
@@ -1 +1,132 @@
...
+  3. Run a volumetric registration algorithm between the imputed volume and the original volume.
+  4. Iterate over the whole dataset until convergence.
+
+
+**Identify an I/O (inputs/outputs) specification**: briefly anticipate what are the inputs to your new algorithm and the expected outcomes.
+
+Inputs
+
+  * A *b=0* reference - this is a 3D file resulting from a varyingly sophisticated average across the *b=0* volumes in the dataset.
+  * Orientation matrix in "RAS+B" format. This means that b-vectors are given in "scanner" coordinates (as opposed to "voxel" coordinates) and must have unit-norm. An additional column provides the sensitization intensity value (*b* value) in *s/mm*.
```
Comment (arokem) [inline docs/dmriprep/hmc.md:47]:
```suggestion
  * Orientation matrix in "RAS+B" format. This means that b-vectors are given in "scanner" coordinates (as opposed to "voxel" coordinates) and must have unit-norm. An additional column provides the sensitization intensity value (*b* value) in *s/mm^2*.
```

File: docs/dmriprep/hmc.md [hunk 8]
```diff
@@ -1 +1,132 @@
...
+  * *high-b* DWI data (4D file) - in other words, the original DWI dataset after extracting the *b=0* volumes out.
+  * A boolean indicating whether this is single-shell, multi-shell or non-shelled (e.g., cartesian sampling such as DSI) dataset.
```
Comment (arokem) [inline docs/dmriprep/hmc.md:49]:
```suggestion
  * A boolean indicating whether this is single-shell, multi-shell or non-shelled (e.g., Cartesian sampling such as DSI) dataset.
```

File: docs/dmriprep/hmc.md [hunk 9]
```diff
@@ -1 +1,132 @@
...
+
+What this idea doesn't cover:
+
+  * Conversion into RAS+B format of the gradient matrix.
+  * Generation of average *b=0* reference.
+  * Calculation of Framewise-Displacement or any other data quality estimation.
+
+**Nonfunctional requirements**: briefly anticipate further requirements that are important, but do not alter the goal of the project.
+
+  * Memory fingerprint: DWIs can be large, and storing them in memory (and subsequent derivatives thereof) can be prohibitive
```
Comment (arokem) [inline docs/dmriprep/hmc.md:69]:
```suggestion
  * Memory fingerprint: DWIs can be large, and storing them in memory (and subsequent derivatives thereof) can be cumbersome, or even prohibitive.
```

**Review by arokem — COMMENTED** [review]
A few small comments and suggestions for now.

File: docs/dmriprep/hmc.md [hunk 10]
```diff
@@ -1 +1,133 @@
-# Head Motion Correction
+# Contributing to *dMRIPrep* - the case of head-motion and eddy-currents estimation
```
Comment (josephmje) [inline docs/dmriprep/hmc.md:1]: Include for rendering python cells.

```suggestion
---
jupytext:
  formats: md:myst
  text_representation:
    extension: .md
    format_name: myst
kernelspec:
  display_name: Python 3
  language: python
  name: python3
---

# Contributing to *dMRIPrep* - the case of head-motion and eddy-currents estimation

```{code-cell} python
:tags: [hide-cell]

import warnings
warnings.filterwarnings("ignore")
```
```

**Comment by oesteban** [comment]
@josephmje, I've pushed some of your edits to `enh/hmc-unit-2` so that the section of this PR that was reviewed by Ariel and Derek could be merged.

**Pull request merged**